DigitalLearningSolutions.Data/DataServices/CourseDataService.cs

```diff
@@ -395,7 +395,7 @@ OUTPUT INSERTED.Id
                  ((SELECT TOP 1 ID FROM SupervisorDelegates WHERE SupervisorAdminID = @supervisorId AND CandidateID = @candidateId), 0) AS ID",
                     new { supervisorId, candidateId }
                 );
-            if (supervisorDelegateId == 0)
+            if (supervisorDelegateId == 0 && supervisorId > 0)
             {
                 supervisorDelegateId = connection.QuerySingle<int>(@"INSERT INTO SupervisorDelegates (SupervisorAdminID, DelegateEmail, CandidateID, SupervisorEmail, AddedByDelegate)

```

Comment:
We should switch this to use `OUTPUT INSERTED.Id` as line 382 instead of `SELECT CAST(SCOPE_IDENTITY() as int)` line 404